aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -144,6 +277,169 @@ impl AptosVM {
         self
     }
 
+    pub(crate) fn new_session<'r, S: AptosMoveResolver>(
+        &self,
+        resolver: &'r S,
+        session_id: SessionId,
+    ) -> SessionExt<'r, '_> {
+        self.move_vm.new_session(resolver, session_id)
+    }
+
+    fn get_gas_parameters(
+        &self,
+        log_context: &AdapterLogSchema,
+    ) -> Result<&AptosGasParameters, VMStatus> {
+        self.gas_params.as_ref().map_err(|err| {
+            let msg = format!("VM Startup Failed. {}", err);
+            speculative_error!(log_context, msg.clone());
+            VMStatus::error(StatusCode::VM_STARTUP_FAILURE, Some(msg))
+        })
+    }
+
+    fn get_storage_gas_parameters(
+        &self,
```

Comment:
gas paramters are stored as `Result` actually, I will adapt it and then we don't even need these functions I believe

Reply:
Let's do that in separate PR as @vgao1996 already approved, as I tried to do it but it is not that straightforward because VM creation can fail now, which has implication on the rest of the system. Changed so we pass &self.storage_gas_params